### DISCUSSION TITLE
Can an interface be opened to allow users to transfer tileSize?

### DISCUSSION THREAD

**Discussion by longhaiyan-lhy** [maplibre/maplibre-gl-js#2944] [Q&A]
---
name: Allows users to transfer tileSize.
about: When `new maplibregl.Map` , tileSize can be transferred, For example, `new maplibregl.Map({tileSize: 256})` . In the subsequent code, the value of tileSize is obtained from the value of tileSize transferred when a user creates a map.
---

## User Story

As a map developer,
when maplibre is used to create a map, tileSize can be adapted based on the window size of different devices.

## Rationale

Currently, the map developed based on maplibre may be used on PCs and mobile devices. Therefore, the tile size needs to be adapted to different ends. Users can also customize tiles when using the map.

## Impact

More personalization

**Comment by HarelM** [comment]
Issues are not meant to ask questions.

There's a template for a reason.

**Comment by longhaiyan-lhy** [comment]
I've added to the issue. Please take a look.

**Reply by HarelM** [reply]
Evry source definition has a tileSize as part of it.
You can and should define the tile size there:
https://maplibre.org/maplibre-style-spec/sources/
